code/__HELPERS/turfs.dm

```diff
@@ -376,6 +376,34 @@ Turf and target are separate in case you want to teleport some distance from a t
 		if (target)
 			return target
 
+///Returns a random department of areas to pass into get_safe_random_station_turf() for more equal spawning.
+/proc/get_safe_random_station_turf_equal_weight()
+	// Big list of departments, each with lists of each area subtype.
+	var/list/department_areas = list(
+				subtypesof(/area/station/engineering), \
+				subtypesof(/area/station/medical), \
+				subtypesof(/area/station/science), \
+				subtypesof(/area/station/security), \
+				subtypesof(/area/station/service), \
+				subtypesof(/area/station/command), \
+				subtypesof(/area/station/hallway), \
+				subtypesof(/area/station/ai_monitored), \
+				subtypesof(/area/station/cargo)
+			)
+
+	var/list/area/final_department = pick(department_areas) // Pick a department
+	var/list/area/final_area_list = list()
+
+	for (var/i = final_department.len, i >= 1, i--) // Check each area to make sure it exists on the station
+		var/area/checked_area = pick_n_take(final_department)
+		if(checked_area in GLOB.the_station_areas)
+			final_area_list += checked_area
```

Comment:
A ) there's no reason for this to be a numeric loop
B ) there's no reason for this to be a reverse iterator

Reply:
It needs to ensure that the areas exist, so it is comparing against GLOB.the_station_areas. Is there a better way to do this?

Reply:
I don't think you get what im saying. 
```
for (var/i = final_department.len, i >= 1, i--)
```
is completely pointless, just do `for(var/path in final_department)`

Reply:
ah I see, 👍